lib/vmpooler/api/v1.rb

```diff
@@ -173,6 +173,69 @@ class V1 < Sinatra::Base
       JSON.pretty_generate(result)
     end
 
+    get "#{api_prefix}/token/:token/?" do
+      content_type :json
+
+      result = { 'ok' => false }
+
+      Vmpooler::API.settings.config[:auth] ? status(401) : status(404)
+
+      if Vmpooler::API.settings.config[:auth] and Vmpooler::API.settings.redis.exists('vmpooler__token__' + params[:token])
+        status(200)
+        result['ok'] = true
+
+        result[params[:token]] = Vmpooler::API.settings.redis.hgetall('vmpooler__token__' + params[:token])        
+      end
+
+      JSON.pretty_generate(result)
+    end
+
+    delete "#{api_prefix}/token/:token/?" do
+      content_type :json
+
+      result = { 'ok' => false }
+
+      Vmpooler::API.settings.config[:auth] ? status(401) : status(404)
+
+      if Vmpooler::API.settings.config[:auth] and Vmpooler::API.settings.redis.exists('vmpooler__token__' + params[:token])
+        status(200)
+        result['ok'] = true
+
+        Vmpooler::API.settings.redis.del('vmpooler__token__' + params[:token])
```

Comment:
`del` returns the number of keys deleted, 0 meaning the key did not exist. This can be setup similar to the `hgetall` comment above.

``` ruby
if config[:auth]
  del_count = redis.del(....)
end

if del_count > 1
  status(200)
  ....
end
```